src/main/java/frc/robot/subsystems/DriveTrain.java

```diff
@@ -116,11 +134,18 @@ public DriveTrain() {
     drive = new DifferentialDrive(leftMaster, rightMaster);
     drive.setSafetyEnabled(false);
     drive.setRightSideInverted(false);
-
-    gyro = new AHRS();
+    try {
+      gyro = new AHRS();
+    } catch (RuntimeException ex) {
+      DriverStation.reportError("Error Instantiating Gyro: " + ex.getMessage(), true);
+    }
 
     odometry = new DifferentialDriveOdometry(Rotation2d.fromDegrees(getHeading()));
 
+    shifter = new DoubleSolenoid(0, 7);
```

Comment:
replace with constants. "use an array with 2 values for double solenoid's ports because better "dustin